File: day01/part1.py
#!/usr/bin/env python3
# vim: set fileencoding=utf-8 :

import fileinput
from bisect import insort, bisect, bisect_right

numlist = [int(line) for line in fileinput.input()]
numlist = sorted(numlist)


def part1(l):
    while l:
        if l[0] + l[-1] == 2020:
            return l[0] * l[-1]
        l = l[:bisect(l, 2020 - l[0])]
        l = l[bisect(l, 2019 - l[-1]):]


print(part1(numlist))
# from timeit import timeit
# print(timeit('part1(numlist)', globals=globals(), number=100000))

File: day02/part1.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $part1 = 0;
while (<>) {
	my ($n1, $n2, $char, $pass) = $_ =~ /(\d+)-(\d+) (.): (.*)/;
	my $count = () = $pass =~ /$char/g; # get the length
	if ($n1 <= $count  and $count <= $n2) {
		$part1++;
	}
}
print "$part1\n";
